moveit_ros/moveit_servo/config/servo_parameters.yaml

```
@@ -191,18 +210,60 @@ servo:
                   this should be set to false"
   }
 
-  halt_all_joints_in_joint_mode: {
+############################### SMOOTHING PLUGIN ###############################
+
+  use_smoothing: {
     type: bool,
+    read_only: true,
     default_value: true,
-    description: "Halt all joints in joint mode"
+    description: "Enables the use of smoothing plugins for joint trajectory smoothing"
   }
 
-  halt_all_joints_in_cartesian_mode: {
+  smoothing_filter_plugin_name: {
+    type: string,
+    read_only: true,
+    default_value: "online_signal_smoothing::ButterworthFilterPlugin",
+    description: "The name of the smoothing plugin to be used"
+  }
+
+############################# COLLISION MONITOR ################################
+
+  check_collisions: {
     type: bool,
     default_value: true,
-    description: "Halt all joints in cartesian mode"
+    description: "If true, servo will check for collision using the planning scene monitor."
+  }
+
+  self_collision_proximity_threshold: {
+    type: double,
+    default_value: 0.01,
+    description: "Start decelerating when a self-collision is this far [m]",
+    validation: {
+      gt<>: 0.0
+    }
+  }
+
+  scene_collision_proximity_threshold: {
+    type: double,
+    default_value: 0.02,
+    description: "Start decelerating when a collision is this far [m]",
+    validation: {
+      gt<>: 0.0
+    }
+  }
+
+  collision_check_rate: {
+    type: double,
```

If you look at `collision_monitor.cpp`, this is only initialized once. Can't be changed at runtime.

```suggestion
  collision_check_rate: {
    type: double,
    read_only: true,
```

---

Actually, the collision check parameters can be updated, since it is initialized locally in the method called in the thread.
The collision checking thread is stopped when Servo is paused. So it is possible to pause Servo, update the params.
When unpaused, collision checking thread will start again with the new params.

That said, the situation is similar to the one we faced with planning group, we lack a mechanism to properly reject updates at the moment, if Servo is not paused.

---

Yes and no... there a few lines at the very top of `CollisionMonitor::checkCollisions()` that do not change:
https://github.com/ros-planning/moveit2/blob/main/moveit_ros/moveit_servo/src/collision_monitor.cpp#L90-L91

Whereas these do react live at runtime:
https://github.com/ros-planning/moveit2/blob/main/moveit_ros/moveit_servo/src/collision_monitor.cpp#L132-L134

So I guess we have a choice of either moving the first lines in the loop (keeping the `-log(0.001)` as a constant), or making the whole thing nontunable. I prefer the former!

---

making them tunable is fine with me. Seems like it shouldn't hurt performance much

---

Yeah, will move the two variable initializations into the loop then.
That should make things nice.

---

Not sure if the change should go in this PR though since this one is about making params `read-only`

---

I think it's small enough (and technically a related bug fix) that you can just throw it in here!